cirq-ft/cirq_ft/infra/gate_with_registers.py

```diff
@@ -26,14 +28,25 @@ class Register:
 
     Args:
         name: The string name of the register
-        bitsize: The number of (qu)bits in the register.
+        shape: Shape of the multi-dimensional qubit register.
     """
 
     name: str
-    bitsize: int
+    shape: Tuple[int, ...] = attr.field(converter=lambda v: (v,) if isinstance(v, int) else v)
+
+    def all_idxs(self) -> Iterable[Tuple[int, ...]]:
+        """Iterate over all possible indices of a multidimensional register."""
+        yield from itertools.product(*[range(sh) for sh in self.shape])
+
+    def total_bits(self) -> int:
+        """The total number of bits in this register.
+
+        This is the product of bitsize and each of the dimensions in `shape`.
+        """
+        return int(np.product(self.shape))
 
     def __repr__(self):
-        return f'cirq_ft.Register("{self.name}", {self.bitsize})'
+        return f'cirq_ft.Register("{self.name}", {self.shape})'
```

Comment:
consider using a kwarg in the repr if you want maximum forward compatibility

Reply:
done